activerecord/lib/active_record/connection_adapters/schema_cache.rb

```diff
@@ -21,22 +22,25 @@ def initialize_dup(other)
         @columns_hash = @columns_hash.dup
         @primary_keys = @primary_keys.dup
         @data_sources = @data_sources.dup
+        @indexes      = @indexes.dup
       end
 
       def encode_with(coder)
-        coder["columns"] = @columns
+        coder["columns"]      = @columns
         coder["columns_hash"] = @columns_hash
         coder["primary_keys"] = @primary_keys
         coder["data_sources"] = @data_sources
-        coder["version"] = connection.migration_context.current_version
+        coder["indexes"]      = @indexes
+        coder["version"]      = connection.migration_context.current_version
       end
 
       def init_with(coder)
-        @columns = coder["columns"]
+        @columns      = coder["columns"]
         @columns_hash = coder["columns_hash"]
         @primary_keys = coder["primary_keys"]
         @data_sources = coder["data_sources"]
-        @version = coder["version"]
+        @indexes      = coder["indexes"]
```

Comment:
This serializes some index objects. I'll define `encode_with` and `init_with` on those too later.

Reply:
Eh, on the other hand maybe we could get away with adding a comment that it's serialized in the schema cache to here: https://github.com/rails/rails/blob/628d3c2489632b6979e0c92eafd4a7ff6c40bfdd/activerecord/lib/active_record/connection_adapters/abstract/schema_definitions.rb#L9-L10